engine/servermaster/executor_manager.go

```diff
@@ -334,7 +371,35 @@ func (e *ExecutorManagerImpl) GetAddr(executorID model.ExecutorID) (string, bool
 		return "", false
 	}
 
-	return executor.Addr, true
+	return executor.Address, true
+}
+
+func (e *ExecutorManagerImpl) resetExecutors(ctx context.Context) error {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	orphanExecutorIDs := make(map[model.ExecutorID]struct{})
+	for id := range e.executors {
+		orphanExecutorIDs[id] = struct{}{}
+	}
+
+	executors, err := e.metaClient.QueryExecutors(ctx)
```

Comment:
If an executor is offline and remains metadata in metastore, it will be added into executor manager and removed after heartbeat timeout. During that time could tasks be dispatched to this nonexistent executor.

Reply:
See https://github.com/pingcap/tiflow/pull/6918#discussion_r957970038.